js/api/Surveys.js

```diff
@@ -6,6 +7,9 @@ import realm from '../data/Realm';
 import { loadForms } from './Forms';
 import { InvitationStatus, loadInvitations, loadCachedInvitation } from '../api/Invitations';
 
+// Live realm.io object containing all cached 'Survey' entries.
+export const realmSurveys = realm.objects('Survey');
```

Comment:
According to realm.io [docs](https://realm.io/docs/react-native/latest/api/Realm.html#objects) this is roughly the equivalent of `SELECT * FROM Surveys` in SQL, which returns all rows from the database. So when this module is imported at runtime, it will load all objects into memory and will remain in memory for the entire life cycle of the application.

This pattern concerns me on mobile devices that have very limited memory footprint. Since we do not (yet) have a concept of pagination and the number of surveys on the remote are kept small so memory isn't a problem.  

Please explain why having all survey objects in memory is useful outside the survey life cycle of the survey list page?

Furthermore, even with this decision to load all the objects into memory, why have not all the methods have been converted to used the in-memory collection within this module?